### PR TITLE
Product Backlog Item #1412942: As an ISV I want to be able to set a label to a file

### PR DESCRIPTION
**Pull request by assapir**
File api with stubs.

### CODE REVIEW

File: sdk/file_sdk/api_impl/stream_handler.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#include "stream_handler.h"
+
+namespace mip {
+namespace file {
+
+StreamHandler::StreamHandler()
```
Comment (vakuras) [inline sdk/file_sdk/api_impl/stream_handler.cpp]: missing the create method #Resolved

File: sdk/file_sdk/api/fileapi_export.h [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#ifndef FILEAPIEXPORT_H
+#define FILEAPIEXPORT_H
+
+#ifdef _WIN32
```
Comment (vakuras) [inline sdk/file_sdk/api/fileapi_export.h:4]: what about unix? #Resolved

File: sdk/file_sdk/api/istream_handler.h [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+  virtual bool IsLabeled() = 0;
+
+  virtual bool IsProtected() = 0;
+
+  virtual std::shared_ptr<Tag> GetLabel() = 0;
+
+  virtual void SetLabel(const std::string& labelId, const LabelingOptions& labelingOptions) = 0;
+
+  virtual void SetLabel(std::shared_ptr<ILabel> label, const LabelingOptions& labelingOptions) = 0;
```
Comment (vakuras) [inline sdk/file_sdk/api/istream_handler.h:34]: you can comment this out until we move to the mip repo, if you don't want to add the label.h class #Resolved

**Comment by vakuras** [comment]
@shenhavba is added to the review. #Closed

**Comment by vakuras** [comment]
:clock1:

File: sdk/file_sdk/api/istream_handler.h (hunk 3 above)
Comment (assapir) [reply sdk/file_sdk/api/istream_handler.h:34]: already added a stub ILabel

---
In reply to: [138427476](https://github.com/AzureAD/rms-sdk-for-cpp/pull/149#discussion_r138427476) [](ancestors = 138427476)

**Review by vakuras — APPROVED** [review]

**Pull request merged**